qa/common/infer_util.py

```diff
@@ -84,474 +90,313 @@ def infer_exact(tester, pf, tensor_shape, batch_size,
+            if len(shm_regions) != 0:
+                if config[1] == "http":
+                    output_req.append(httpclient.InferRequestedOutput(
+                        OUTPUT1, binary_data=config[3]))
+                else:
+                    output_req.append(grpcclient.InferRequestedOutput(OUTPUT1))
+
+                if precreated_shm_regions is None:
+                    output_req[-1].set_shared_memory_region(
+                        shm_regions[2+i]+'_data', output1_byte_size)
+                else:
+                    output_req[-1].set_shared_memory_region(
+                        precreated_shm_regions[i], output1_byte_size)
             else:
                 if output1_raw:
-                    output_req[OUTPUT1] = InferContext.ResultFormat.RAW
+                    if config[1] == "http":
+                        output_req.append(httpclient.InferRequestedOutput(
+                            OUTPUT1, binary_data=config[3]))
+                    else:
+                        output_req.append(
+                            grpcclient.InferRequestedOutput(OUTPUT1))
                 else:
-                    output_req[OUTPUT1] = (InferContext.ResultFormat.CLASS, num_classes)
-
-        if len(shm_handles) != 0:
-            results = ctx.run(
-                    { INPUT0 : (shm_handles[0], tensor_shape),
-                    INPUT1 : (shm_handles[1], tensor_shape) },
-                    output_req, batch_size,
-                    priority=priority, timeout_us=timeout_us)
+                    if config[1] == "http":
+                        output_req.append(httpclient.InferRequestedOutput(
+                            OUTPUT1, binary_data=False, class_count=num_classes))
+                    else:
+                        output_req.append(grpcclient.InferRequestedOutput(
+                            OUTPUT1, class_count=num_classes))
+
+        if model_version is not None:
+            model_version = str(model_version)
+        else:
+            model_version = ""
+
+        if config[2]:
+            # TODO fix for streaming case
+            continue
+            # results = triton_client.async_stream_infer(model_name,
+            #                                  inputs,
+            #                                  model_version=model_version,
+            #                                  stream=stream,
+            #                                  outputs=output_req)
         else:
-            results = ctx.run(
-                    { INPUT0 : input0_list, INPUT1 : input1_list },
-                    output_req, batch_size,
-                    priority=priority, timeout_us=timeout_us)
+            results = triton_client.infer(model_name,
+                                        inputs,
+                                        model_version=model_version,
+                                        outputs=output_req,
+                                        request_id=str(_unique_request_id()))
```

Comment:
Did the V1 library automatically assign a request ID? Is that why you added it here explicitly because V2 does not?

Reply:
That is correct